agent/replacecall/src/main/java/randoop/instrument/ReplaceCallAgent.java

```diff
@@ -260,6 +349,121 @@ private static Set<String> loadExclusions(@Owning Reader exclusionReader, String
     return excludedPackagePrefixes;
   }
 
+  /**
+   * Parses a string into a {@link MethodSignature}.
+   *
+   * <p>Accepts the colon/JVM-descriptor form: {@code pkg.Clazz.method:(...)}. Examples: {@code
+   * java.lang.System.exit:(I)V}, {@code
+   * java.util.Objects.requireNonNull:(Ljava/lang/Object;)Ljava/lang/Object;}
+   *
+   * <p>Returns null if parsing fails.
+   *
+   * @param text the string to parse
+   * @return the parsed {@link MethodSignature} or null if parsing fails
+   */
+  private static @Nullable MethodSignature parseMethodSignature(String text) {
+    String s = text.trim();
+
+    int colonPos = s.indexOf(':');
+    if (colonPos > 0 && s.indexOf('(') > colonPos) {
+      String fqName = s.substring(0, colonPos);
+      int dotPos = fqName.lastIndexOf('.');
+      if (dotPos > 0) {
+        String owner = fqName.substring(0, dotPos);
+        String name = fqName.substring(dotPos + 1);
```

Comment:
Why bother parsing `fqName` into `owner` and `name`, given that the only way they are used is being concatenated back together, exactly like `fqName`?